nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -776,6 +779,23 @@ protected Collection<ValidationResult> computeValidationErrors(final ValidationC
             final Collection<ValidationResult> referencedServiceValidationResults = validateReferencedControllerServices(validationContext);
             validationResults.addAll(referencedServiceValidationResults);
 
+            analyze();
+
+            Optional.ofNullable(getValidationContextFactory().getRuleViolationsManager())
+                .map(ruleViolationsManager -> ruleViolationsManager.getRuleViolationsForSubject(getIdentifier()))
+                .map(Collection::stream)
+                .ifPresent(ruleViolationStream -> ruleViolationStream
+                    .filter(ruleViolation -> ruleViolation.getEnforcementPolicy() == EnforcementPolicy.ENFORCE)
+                    .filter(RuleViolation::isEnabled)
+                    .forEach(ruleViolation -> validationResults.add(
+                        new ValidationResult.Builder()
+                            .subject(getComponent().getClass().getSimpleName())
+                            .valid(false)
+                            .explanation(ruleViolation.getViolationMessage())
+                            .build()
+                    )
+                ));
```

Comment:
We should avoid big, nested blocks of functional code. This would be much cleaner and easier to maintain in an imperative form.

Reply:
I can change it.